### ISSUE TITLE
When attempting to start a netgame from map number MAP33 or greater, the chocolate-doom.exe and chocolate-doom-setup.exe remain hanging as processes but having no windows.

### ISSUE THREAD

**Issue by Zodomaniac** [chocolate-doom/chocolate-doom#875]
IMO the engine should not hang in this case but exit with an error message.
Here's the WAD I made from GothicDM duplicating some maps to test this. Remove the additional .txt extension, I added it because GitHub says it doesn't support ZIPs uploaded from certain machines.
[GOTHICDM33.zip.txt](https://github.com/chocolate-doom/chocolate-doom/files/810160/GOTHICDM33.zip.txt)

**Comment by fabiangreffrath** [comment]
The question is if we should error out if `NET_ValidGameSettings()` returns `false`. The current strategy of not doing anything somehow appears wrong to me.

**Comment by Zodomaniac** [comment]
@fabiangreffrath 
I can't see any reason why not error out, it would be natural and logical, I'm really glad that this change is pending :) given that there's even a dedicated function for that, `NET_ValidGameSettings()` you mentioned.
If netgame setting are invalid, error out, yes, don't hang!

**Comment by fabiangreffrath** [comment]
This would be here: https://github.com/chocolate-doom/chocolate-doom/blob/sdl2-branch/src/net_server.c#L949

**Comment by Zodomaniac** [comment]
So what kind of solution do we agree on?

**Comment by fabiangreffrath** [comment]
How about this?

```
--- a/src/net_server.c
+++ b/src/net_server.c
@@ -946,6 +946,7 @@ static void NET_SV_ParseGameStart(net_packet_t *packet, net_client_t *client)

         if (!NET_ValidGameSettings(sv_gamemode, sv_gamemission, &settings))
         {
+            NET_SV_SendReject(addr, "Invalid game settings detected!\n");
             return;
         }

```

**Comment by Zodomaniac** [comment]
@fabiangreffrath 
> How about this

Nice idea IMO)

**Comment by fragglet** [comment]
As I recall, NET_SV_SendReject is for rejecting clients joining the server
and not appropriate for clients who have already connected.

Isn't there a message that lets the server send a print command to a
client? That would make more sense.

**Comment by fabiangreffrath** [comment]
But we don't only want to send a message, we also want to disconnect from this client. Or not?

**Comment by fabiangreffrath** [comment]
> we also want to disconnect from this client

So, we could call a `NET_SV_GameEnded()` after sending the message, right?

Please, I need a pointer, I am really not familiar enough with the networking code.

**Comment by fragglet** [comment]
Can't end a game before it's started! I suggest a `NET_SV_SendConsoleMessage()` possibly followed by a `NET_SV_DisconnectClient()`.

I'm slightly perplexed as to why the game hangs if the game settings are invalid though. It makes sense that the server would not start the game, but it doesn't make sense that either the client or server should freeze up.

**Comment by fabiangreffrath** [comment]
I think this is because if `NET_ValidGameSettings()` returns `false` then `NET_SV_ParseGameStart()` merely `return`s without processing any of the calls further down the code.

**Comment by fragglet** [comment]
Yeah, I don't see why that would freeze the game though. Something seems odd.

**Comment by mfrancis95** [comment]
I tried a `NET_SV_SendConsoleMessage()` followed by a `NET_SV_DisconnectClient`. When testing with a dedicated server and one client, the client will get disconnected but never receive the message. When using one client with `-privateserver` and another client, both connections will seemingly close, but an instance of Choco still gets started up.

I'm wondering if one way to deal with this would be to create a new packet type that is sent when the game settings are invalid.

**Comment by fabiangreffrath** [comment]
Any news on this?

**Comment by fabiangreffrath** [comment]
I tried this patch and `chocolate-doom -privateserver -episode 2` and the game ended with a "Lost connection to server" messagebox. Well, at least that is something and it doesn't hang anymore.

```
--- a/src/net_server.c
+++ b/src/net_server.c
@@ -981,6 +981,8 @@ static void NET_SV_ParseGameStart(net_packet_t *packet, net_client_t *client)
         if (!NET_ValidGameSettings(sv_gamemode, sv_gamemission, &settings))
         {
             NET_Log("server: error: invalid game settings");
+            NET_SV_SendConsoleMessage(client, "server: error: invalid game settings");
+            NET_SV_DisconnectClient(client);
             return;
         }

```

**Comment by mfrancis95** [comment]
Thoughts on this?

> I'm wondering if one way to deal with this would be to create a new packet type that is sent when the game settings are invalid.

**Comment by Zodomaniac** [comment]
I think Choco 3.0.1 can be released when this is fixed :)

**Comment by Zodomaniac** [comment]
I think I have a reason to resurrect this topic requesting changes that would make forks able to run multiplayer MAP33/E1M10/NRFTL/SIGIL. Choco should handle these maps in some other way than hanging, as it's been agreed upon above.
